lib/game_logic.rb

```diff
@@ -0,0 +1,54 @@
+class Board
+  attr_accessor :valid, :square
+  attr_reader :game
+
+  def initialize
+    @square = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    @game = true
+    @valid = false
+  end
+
+  def create_board
+    puts '-----+-----+-----'
+    puts "[ #{@square[0]} ] [ #{@square[1]} ] [ #{@square[2]} ]"
+    puts "[ #{@square[3]} ] [ #{@square[4]} ] [ #{@square[5]} ]"
+    puts "[ #{@square[6]} ] [ #{@square[7]} ] [ #{@square[8]} ]"
+    puts '-----+-----+-----'
+  end
+
+  def check_valid_move(input)
+    @input = input
+    if !@square.include?(input)
+      system 'cls'
+      puts 'Please use a valid number'
+      sleep 1
+      @valid = false
+    else
+      @valid = true
+    end
+  end
+
+  def win_check(arr, letter)
+    return true if [@square[arr[0]], @square[arr[1]], @square[arr[2]]].all?(letter)
```

Comment:
## [Optional] 
you can make `win_check` method as private as it is only calling from another method of the same class.
This is optional but I will highly recommend it.